### PR TITLE
Fix up some pytest style issues

### PR DESCRIPTION
**Pull request by akx**
This PR fixes tests to use `pytest.raises` and `pytest.warns` as context managers everywhere possible.

### CODE REVIEW

File: Tests/test_core_resources.py [hunk 1]
```diff
@@ -177,13 +177,14 @@ def test_units(self):
...
+        "vars",
+        [
+            {"PILLOW_ALIGNMENT": "15"},
+            {"PILLOW_BLOCK_SIZE": "1024"},
+            {"PILLOW_BLOCKS_MAX": "wat"},
+        ],
+    )
+    def test_warnings(self, vars):
+        with pytest.warns(UserWarning):
+            Image._apply_env_variables(vars)
```
Comment (hugovk) [inline Tests/test_core_resources.py:190]: Singular, only applying one at a time:
```suggestion
        "var",
        [
            {"PILLOW_ALIGNMENT": "15"},
            {"PILLOW_BLOCK_SIZE": "1024"},
            {"PILLOW_BLOCKS_MAX": "wat"},
        ],
    )
    def test_warnings(self, var):
        with pytest.warns(UserWarning):
            Image._apply_env_variables(var)
```

File: Tests/test_file_tiff_metadata.py [hunk 2]
```diff
@@ -422,7 +423,8 @@ def test_too_many_entries():
     ifd.tagtype[277] = TiffTags.SHORT
 
     # Should not raise ValueError.
-    pytest.warns(UserWarning, lambda: ifd[277])
+    with pytest.warns(UserWarning):
+        _ = ifd[277]
```
Comment (radarhere) [inline Tests/test_file_tiff_metadata.py:427]:
```suggestion
        ifd[277]
```

File: Tests/test_file_tiff.py [hunk 3]
```diff
@@ -57,11 +57,10 @@ def test_sanity(self, tmp_path):
 
     @pytest.mark.skipif(is_pypy(), reason="Requires CPython")
     def test_unclosed_file(self):
-        def open():
+        with pytest.warns(ResourceWarning):
             im = Image.open("Tests/images/multipage.tiff")
             im.load()
-
-        pytest.warns(ResourceWarning, open)
+            del im
```
Comment (radarhere) [inline Tests/test_file_tiff.py:63]: Why did you add this? Users should no longer delete images - https://pillow.readthedocs.io/en/stable/deprecations.html#image-del

File: Tests/test_file_tiff_metadata.py (hunk 2 above)
Comment (nulano) [reply Tests/test_file_tiff_metadata.py:427]: `idf[277]` shows a `Statement seems to have no effect` warning in PyCharm, while `_ = ifd[277]` does not.

Comment (akx) [reply Tests/test_file_tiff_metadata.py:427]: What @nulano said – I didn't want the statement to look like a no-op (even if it has a side effect of raising a warning) and `_` is a common placeholder name, so I figured the intent is pretty clear.

File: Tests/test_file_tiff.py (hunk 3 above)
Comment (akx) [reply Tests/test_file_tiff.py:63]: Because if the image is not `del`eted (or otherwise GC'd) without having been closed, the resource warning isn't triggered.

With the previous `lambda` syntax, the variable would've dropped out of scope and had the same effect.

File: Tests/test_file_tiff_metadata.py (hunk 2 above)
Comment (radarhere) [reply Tests/test_file_tiff_metadata.py:427]: Ok. What about asserting the value instead? https://github.com/akx/Pillow/pull/1

Comment (akx) [reply Tests/test_file_tiff_metadata.py]: @radarhere Sure, that works! Merged your PR, cheers.

**Comment by radarhere** [comment]
Is there a link that you can provide, describing this pytest style preference?

**Comment by akx** [comment]
@radarhere Kind of:

```
(Pillow) ~/b/Pillow (main) $ git grep pytest.raises | grep with | wc -l
     505
(Pillow) ~/b/Pillow (main) $ git grep pytest.raises | grep -v with | wc -l
      18
(Pillow) ~/b/Pillow (main) $ git grep pytest.warns | grep with | wc -l
      48
(Pillow) ~/b/Pillow (main) $ git grep pytest.warns | grep -v with | wc -l
      27
```
IOW, Pillow's tests themselves overwhelmingly use `with pytest.raises`/`with pytest.warns`.

**Comment by hugovk** [comment]
And I find `im.save(out, save_all=True, append_images=ims)` cleaner/more readable and closer to normal use than a big list of parameters: `UserWarning, im.save, out, save_all=True, append_images=ims`.

File: Tests/test_file_apng.py [hunk 4]
```diff
@@ -263,12 +263,9 @@ def test_apng_chunk_errors():
     with Image.open("Tests/images/apng/chunk_no_actl.png") as im:
         assert not im.is_animated
 
-    def open():
+    with pytest.warns(UserWarning):
         with Image.open("Tests/images/apng/chunk_multi_actl.png") as im:
             im.load()
```
Comment (radarhere) [inline Tests/test_file_apng.py:268]:
```suggestion
        with Image.open("Tests/images/apng/chunk_multi_actl.png") as im:
            im.load()
        assert not im.is_animated
```
Is there any particular reason you removed the `assert`?

File: Tests/test_file_tiff.py (hunk 3 above)
Comment (radarhere) [reply Tests/test_file_tiff.py:63]: What about https://github.com/akx/Pillow/pull/4 instead?

Comment (akx) [reply Tests/test_file_tiff.py]: Fine by me. Merged that in here.

File: Tests/test_file_apng.py (hunk 4 above)
Comment (akx) [reply Tests/test_file_apng.py:268]: Accidentally :)

File: Tests/test_core_resources.py [hunk 5]
```diff
@@ -177,13 +177,14 @@ def test_units(self):
...
-        pytest.warns(
-            UserWarning, Image._apply_env_variables, {"PILLOW_BLOCKS_MAX": "wat"}
-        )
+    @pytest.mark.parametrize(
+        "var",
+        [
+            {"PILLOW_ALIGNMENT": "15"},
+            {"PILLOW_BLOCK_SIZE": "1024"},
+            {"PILLOW_BLOCKS_MAX": "wat"},
+        ],
```
Comment (radarhere) [inline Tests/test_core_resources.py:186]:
```suggestion
        (
            {"PILLOW_ALIGNMENT": "15"},
            {"PILLOW_BLOCK_SIZE": "1024"},
            {"PILLOW_BLOCKS_MAX": "wat"},
        ),
```

**Review by radarhere — APPROVED** [review]

File: Tests/test_file_webp.py [hunk 6]
```diff
@@ -29,7 +29,9 @@ def test_unsupported(self):
             WebPImagePlugin.SUPPORTED = False
 
         file_path = "Tests/images/hopper.webp"
-        pytest.warns(UserWarning, lambda: pytest.raises(OSError, Image.open, file_path))
+        with pytest.warns(UserWarning):
+            with pytest.raises(OSError):
+                Image.open(file_path)
```
Comment (radarhere) [inline Tests/test_file_webp.py:34]:
```suggestion
                with Image.open(file_path):
                    pass
```
